src/test/groovy/com/user_service/services/UserAuthenticationContractTest.groovy

```diff
@@ -0,0 +1,148 @@
+package com.user_service.services
+
+import com.user_service.controllers.UserAuthenticationController
+import com.user_service.dtos.request.*
+import com.user_service.dtos.response.*
+import org.springframework.http.HttpStatus
+import spock.lang.Specification
+
+import java.time.LocalDate
+
+class UserAuthenticationContractTest extends Specification {
+
+    UserAuthenticationController userAuthenticationController
+    UserAuthenticationService userAuthenticationService
+
+    def setup() {
+        userAuthenticationService = Mock(UserAuthenticationService)
```

Comment:
I think you mock too much. You should mock either `ExternalCognitoClient` or `CognitoIdentityProviderClient` itself. As you noticed, the coverage of tests is very low, because everything is mocked and you don't test a lot of logic.